### PR TITLE
feature: test and deploy with github

### PR DESCRIPTION
**Pull request by unRob**
## Context

Turns out it's really easy to publish docs to github via actions, and doing so from CircleCI requires a bit more fine-tuning

## Description of changes

- Publish and deploy npm package with github actions
- test in github actions
- produce lcov coverage, remove junit plugin
- push coverage report to [coveralls.io](https://coveralls.io/github/blinkhealth/config-yourself-javascript)
- add test and coverage badges to readme

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,13 +22,13 @@
     "src"
   ],
   "engines": {
-    "node": ">=8.12.0"
+    "node": ">=10.9.0"
```
Comment (Ikana) [inline package.json]: Why not using node 12?

Comment (unRob) [reply package.json]: If I understand it correctly, that specifies the minimum version that this package will work with. It builds and runs with 12.

Comment (unRob) [reply package.json]: I guess I'd be more convinced if I understood the advantages of restricting its usage to later versions

Comment (Ikana) [reply package.json]: Yes, sorry LGTM, I got the other node stuff mixed up on my head this is a library it does not matter 😅

**Pull request merged**